ipatests/test_integration/test_automember.py

```diff
@@ -0,0 +1,366 @@
+            self.user_automember_cleanup(user1)
+
+            result = self.master.run_command(
+                ['ipa', 'group-show', group2]
+            )
+            assert user1 not in result.stdout_text
+
+        finally:
+            # testcase cleanup
+            self.automember_group_cleanup(group1)
+            self.automember_group_cleanup(group2)
+
+    def test_modify_host_entry_unmembership(self):
+        # https://pagure.io/389-ds-base/issue/50077
+        """
+        Test that an update on a host entry triggers a re-evaluation of the
+        host's automember groups when auto-unmembership is active.
+        """
+        try:
+            # testcase setup
+            hostgroup1 = 'Brno'
+            hostgroup2 = 'Boston'
+            host1 = 'host1.example.com'
+
+            self.automember_hostgroup_add(hostgroup1)
+            self.automember_hostgroup_add(hostgroup2)
+            self.host_automember_setup(hostgroup1, host1)
+
+            result = self.master.run_command(
+                ['ipa', 'hostgroup-show', hostgroup1]
+            )
+            assert host1 in result.stdout_text
+
+            # Modifying the user group so that it becomes part of new group
+            self.host_automember_mod(hostgroup2, host1)
+
+            result = self.master.run_command(
+                ['ipa', 'hostgroup-show', hostgroup2]
+            )
+            assert host1 in result.stdout_text
+
+            result = self.master.run_command(
+                ['ipa', 'hostgroup-show', hostgroup1]
+            )
+            assert host1 not in result.stdout_text
+
+            # Deleting the user entry and checking its not part of group now
+            self.host_automember_cleanup(host1)
+
+            result = self.master.run_command(
+                ['ipa', 'hostgroup-show', hostgroup2]
+            )
+            assert host1 not in result.stdout_text
+
+        finally:
+            # testcase cleanup
+            self.automember_hostgroup_cleanup(hostgroup1)
+            self.automember_hostgroup_cleanup(hostgroup2)
+
+    def test_modify_user_entry_unmembership_disabled(self):
+        # https://pagure.io/389-ds-base/issue/50077
+        """
+        Test that an update on a user entry does not triggers re-evaluation of
+        the user's automember groups when auto-unmembership is disabled. We
+        need to run automembership-rebuild for correct membership of user.
+        """
+        try:
+            master = self.master
+            tf = NamedTemporaryFile()
+            ldif_file = tf.name
+
+            entry_ldif = textwrap.dedent("""
+                dn: cn=Auto Membership Plugin,cn=plugins,cn=config
+                changetype: modify
+                replace: automemberprocessmodifyops
+                automemberprocessmodifyops: off""")
+
+            master.put_file_contents(ldif_file, entry_ldif)
+
+            arg = ['ldapmodify',
```

Comment:
Instead of calling ldapmodify on master it is better to use update_entry() method of ldap client provided by  master.ldap_connect. You can see axample of usage in test_permission_plugin. This way you will not need to create and transfer the diff file.